docs/rules/no-mixed-operators.md

```diff
@@ -0,0 +1,125 @@
+# Disallow mixes of different operators (no-mixed-operators)
+
+Enclosing complex expressions by parentheses makes developer's intention clarifying, then the code would get more readability.
+This rule warns that there are different operators consecutively without parentheses *in AST*.
+
+```js
+var foo = a && b || c || d;    /*BAD: Unexpected a mix of '&&' and '||'.*/
+var foo = (a && b) || c || d;  /*GOOD*/
+var foo = a && (b || c || d);  /*GOOD*/
+```
+
+## Rule Details
+
+This rule checks `BinaryExpression` and `LogicalExpression`.
+
+This rule would conflict with [no-extra-parens] rule.
+If you use both this and [no-extra-parens] rule together, you need to use `nestedBinaryExpressions` option of [no-extra-parens] rule.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint no-mixed-operators: "error"*/
+
+var foo = a && b < 0 || c > 0 || d + 1 === 0;
+var foo = a + b * c;
+```
+
+Examples of **correct** code for this rule:
+
+```js
+/*eslint no-mixed-operators: "error"*/
+
+var foo = a || b || c;
+var foo = a && b && c;
+var foo = (a && (b < 0)) || (c > 0) || ((d + 1) === 0);
+var foo = a && ((b < 0) || (c > 0) || ((d + 1) === 0));
+var foo = a + (b * c);
+var foo = (a + b) * c;
+```
+
+## Options
+
+```json
+{
+    "no-mixed-operators": [
+        "error",
+        {
+            "groups": [],
+            "ignoreSamePrecedence": false
+        }
+    ]
+}
+```
+
+This rule has 2 options.
+
+* `groups` (`string[][]`) - specifies groups to compare operators.
+  When this rule compares two operators, if both operators are included in a same group, this rule checks it. Otherwise, this rule ignores it.
+  This value is a list of groups. The group is a list of binary operators.
+  Default is a group which includes all operators.
+* `ignoreSamePrecedence` (`boolean`) - specifies to ignore a mix of 2 operators if those have the same precedence. Default is `false`.
+
+### groups
+
+The following operators can be used in `groups` option:
+
+* Arithmetic Operators: `"+"`, `"-"`, `"*"`, `"/"`, `"%"`, `"**"`
+* Bitwise Operators: `"&"`, `"|"`, `"^"`, `"~"`, `"<<"`, `">>"`, `">>>"`
+* Comparison Operators: `"=="`, `"!="`, `"==="`, `"!=="`, `">"`, `">="`, `"<"`, `"<="`
+* Logical Operators: `"&&"`, `"||"`
+* Relational Operators: `"in"`, `"instanceof"`
+
+Default is a group which includes all operators, i.e. `{"groups": [["+", "-", "*", "/", "%", "**", "&", "|", "^", "~", "<<", ">>", ">>>", "==", "!=", "===", "!==", ">", ">=", "<", "<=", "&&", "||", "in", "instanceof"]]}`.
```

Comment:
Though my preference is only `["&&", "||"]` :smile: 
What about:

Default is the groups for each kind:

    {"groups": [
        ["+", "-", "*", "/", "%", "**"], 
        ["&", "|", "^", "~", "<<", ">>", ">>>"], 
        ["==", "!=", "===", "!==", ">", ">=", "<", "<="], 
        ["&&", "||"], 
        ["in", "instanceof"]
    ]}